package MsgsLauncherToAdmin;

import "Common.proto";

message MsgConnectionAck
{
    enum ErrorCode {
        success							= 0;
        invalid_connection				= 1;
        incompatible_protocol_version	= 2;
    }
    required ErrorCode  error_code		= 1;
    optional int32		context			= 2;
}

message MsgExercicesListResponse
{
    enum ErrorCode {
        success				= 0;
        failure				= 1;
    }
    required ErrorCode  error_code		= 1;
	repeated string exercise_name		= 2;
}

message MsgControlStartAck 
{
    enum ErrorCode {
        success				= 0;
        bad_exercise_name	= 1;
    }
    required ErrorCode  error_code		= 1;
    required string exercise_name		= 2;
}

message MsgControlPauseAck 
{
    enum ErrorCode {
        success				= 0;
        bad_exercise_name	= 1;
    }
    required ErrorCode  error_code		= 1;
    required string exercise_name		= 2;
}

message MsgControlStopAck 
{
    enum ErrorCode {
        success				= 0;
        bad_exercise_name	= 1;
    }
    required ErrorCode  error_code		= 1;
    required string exercise_name		= 2;
}